src/zx_graph.jl

```diff
@@ -144,31 +184,34 @@ end
 rem_spider!(zxg::ZXGraph{T, P}, v::T) where {T, P} = rem_spiders!(zxg, [v])
 
 function add_spider!(zxg::ZXGraph{T, P}, st::SpiderType.SType, phase::P = zero(P), connect::Vector{T}=T[]) where {T<:Integer, P}
-    add_vertex!(zxg.mg)
-    v = vertices(zxg.mg)[end]
-    zxg.ps[v] = phase
-    zxg.st[v] = st
+    v = add_vertex!(zxg.mg)[]
```

Comment:
```suggestion
    v = add_vertex!(zxg.mg)[1]
```

Reply:
I think we are sure we only want the first element (and it will only contain one element) `[]` will check if it is a one-element `Vector`, we do not need this check I think.